src/core/filesystem.cc

```diff
@@ -1042,7 +1044,7 @@ GetFileSystem(const std::string& path, FileSystem** file_system)
         "-DTRTIS_ENABLE_S3=ON.");
 #else
     re2::RE2 s3_regex(
-        "s3://([a-z]+):([0-9]+)/([0-9a-z.-]+)(((/[0-9a-zA-Z.-_]+)*)?)");
+        "s3://([0-9a-zA-Z-.]+):([0-9]+)/([0-9a-z.-]+)(((/[0-9a-zA-Z.-_]+)*)?)");
```

Comment:
You still have this s3 regex repeated here, when you should be able to have it only in the s3 filesystem object. The following logic should be moved into the S3Filesystem constructor.

Reply:
The s3 filesystem object is created once we know the host and port number since the configuration is done inside the constructor. 
Are you suggesting I pull this outside the constructor into a different function?

Reply:
I'm suggesting we don't want to have to create an s3_regex outside of the class. We know this is an s3 path so just pass the path to the constructor and let it do what it wants (find host and port, etc.).

Reply:
Done